src/main/java/de/tum/in/www1/artemis/web/rest/repository/TestRepositoryResource.java

```diff
@@ -40,12 +40,47 @@ public TestRepositoryResource(UserService userService, AuthorizationCheckService
         this.versionControlService = versionControlService;
     }
 
+    /**
+     * Retrieve a test repository by providing its exerciseId. Will check if the user has permissions to access data related to the given test repository.
+     *
+     * @param exerciseId of the given test repository's exercise.
+     * @param pullOnGet  perform a pull on retrieval of a git repository (in some cases it might make sense not to pull!)
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws GitAPIException
+     */
     @Override
-    Repository getRepository(Long exerciseId) throws IOException, IllegalAccessException, InterruptedException {
+    Repository getRepository(Long exerciseId, boolean pullOnGet) throws IOException, IllegalAccessException, InterruptedException, GitAPIException {
         ProgrammingExercise exercise = (ProgrammingExercise) exerciseService.findOne(exerciseId);
         String testRepoName = exercise.getProjectKey().toLowerCase() + "-tests";
         URL testsRepoUrl = versionControlService.get().getCloneURL(exercise.getProjectKey(), testRepoName);
-        return repositoryService.checkoutRepositoryByName(exercise, testsRepoUrl);
+        return repositoryService.checkoutRepositoryByName(exercise, testsRepoUrl, pullOnGet);
+    }
+
+    /**
+     * Get the test repository url by providing a exercise id. Will not check any permissions!
+     *
+     * @param exerciseId
+     * @return
+     */
+    @Override
+    URL getRepositoryUrl(Long exerciseId) {
+        ProgrammingExercise exercise = (ProgrammingExercise) exerciseService.findOne(exerciseId);
+        String testRepoName = exercise.getProjectKey().toLowerCase() + "-tests";
```

Comment:
this assumes that the test repo always ends with "-tests". I think we store the testRepoUrl in the exercise. Could we not simply use this url here?

Reply:
You are right, I probably copied it from the area where the test repository is created.
I implemented a class member `getTestRepositoryName` that extracts the name form the url.